kernel/sched.c

```diff
@@ -34,12 +34,34 @@ static void update_cache(int preempt_ok);
 static void halt_thread(struct k_thread *thread, uint8_t new_state);
 static void add_to_waitq_locked(struct k_thread *thread, _wait_q_t *wait_q);
 
-
 BUILD_ASSERT(CONFIG_NUM_COOP_PRIORITIES >= CONFIG_NUM_METAIRQ_PRIORITIES,
 	     "You need to provide at least as many CONFIG_NUM_COOP_PRIORITIES as "
 	     "CONFIG_NUM_METAIRQ_PRIORITIES as Meta IRQs are just a special class of cooperative "
 	     "threads.");
 
+#ifdef CONFIG_SCHED_DEADLINE
+/*
+ * Compare two deadlines
+ *
+ * > 0 -> d1 occurs before d2
+ * = 0 -> d1 and d2 are the same deadline
+ * < 0 -> d1 occurs after d2
+ */
+int32_t z_sched_deadline_cmp(uint32_t d1, uint32_t d2)
```

Comment:
Should be `static ALWAYS_INLINE`, no?   This logic is too subtle to express as an "API".  Factoring it out to eliminate duplication is fine, but let's load as few footguns as possible.

Reply:
Will do.